docs/design.md

```diff
@@ -431,3 +432,106 @@ that don't use PMEM-CSI at all.
 Users must take care to create PVCs first, then the pods if they want
 to use the webhook. In practice, that is often already done because it
 is more natural, so it is not a big limitation.
+
+### Operator API
+
+PMEM-CSI operator facilitates deploying and managing the [PMEM-CSI driver](https://github.com/intel/pmem-csi) on a Kubernetes cluster. 
+This operator is based on the CoreOS [operator-sdk](https://github.com/operator-framework/operator-sdk) tools and APIs.
+This section describes the custom resource definition API provided for deploying the PMEM-CSI driver.
+
+#### Deployment CRD API
+
+`Deployment` is a cluster-scoped Kubernetes resource in the
+`pmem-csi.intel.com` API group. It describes how a PMEM-CSI driver
+instance is to be created.
+
+The operator will create objects in the namespace in which the
+operator itself runs unless the object type is not namespaced.
+
+The name of the deployment object is also used as CSI driver
+name. This ensures that the name is unique and immutable. However,
+name clashes with other CSI drivers are still possible, so the name
+should meet the [CSI
+requirements](https://github.com/container-storage-interface/spec/blob/master/spec.md#getplugininfo):
+- domain name notation format, including a unique top-level domain
+- 63 characters or less, beginning and ending with an alphanumeric
+  character ([a-z0-9A-Z]) with dashes (-), dots (.), and
+  alphanumerics between.
+
+The name is also used as prefix for the names of all objects created
+for the deployment and for the local `/var/lib/<name>` state directory
+on each node.
+
+The current API for PMEM-CSI `Deployment` resources is:
+
+#### Deployment
+
+|Field | type | Description |
+|---|---|---|
+| apiVersion | string  | `pmem-csi.intel.com/v1alpha1`|
+| kind | string | `Deployment`|
+| metadata | [ObjectMeta](https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status) | Object metadata, name used for CSI driver and as prefix for sub-objects |
```

Comment:
This link "spec-and-status" currently shows 404 page not found

Reply:
Don't link to "devel", use a fixed tag or hash.

Reply:
There are no tags, replaced the link with `master`. Is it ok?